.gitlab-ci.yml

```diff
@@ -524,53 +527,47 @@ check_with_odd_no_of_processors:
-#    - linux
-#    - icp
-
-#deploy_sphinx_documentation:
-#  extends: .deploy_base
-#  dependencies:
-#    - check_sphinx
-#  script:
-#    - cd ${CI_PROJECT_DIR}/build/doc/sphinx/html &&
-#      rsync -avz --delete -e "ssh -i ${HOME}/.ssh/espresso_rsa" ./ espresso@elk.icp.uni-stuttgart.de:/home/espresso/public_html/html/doc
-
-#deploy_doxygen_documentation:
-#  extends: .deploy_base
-#  dependencies:
-#    - run_doxygen
-#  script:
-#    - cd ${CI_PROJECT_DIR}/build/doc/doxygen/html &&
-#      rsync -avz --delete -e "ssh -i ${HOME}/.ssh/espresso_rsa" ./ espresso@elk.icp.uni-stuttgart.de:/home/espresso/public_html/html/dox
-
-#deploy_tutorials:
-#  extends: .deploy_base
-#  dependencies:
-#    - run_tutorials
-#  script:
-#    - cd ${CI_PROJECT_DIR}/build/doc/tutorials
-#    - ${CI_PROJECT_DIR}/maintainer/CI/deploy_tutorials.py
-#    - mkdir www &&
-#      rsync -a --files-from=deploy_list.txt ./ www &&
-#      rsync -avz --delete -e "ssh -i ${HOME}/.ssh/espresso_rsa" www/ espresso@elk.icp.uni-stuttgart.de:/home/espresso/public_html/html/tutorials_html
-#  only:
-#    - schedules
+deploy_documentation:
+  <<: *global_job_definition
+  stage: deploy
+  only:
+    - python
+    - schedules
+  dependencies:
+    - check_sphinx
+    - run_doxygen
+    - run_tutorials
+  before_script:
+    # https://docs.github.com/en/github/authenticating-to-github/githubs-ssh-key-fingerprints
+    - 'which ssh-agent && eval $(ssh-agent -s) || exit 1'
+    - ssh-add <(echo "${GITHUB_PAGES_SSH_PRIVATE_KEY}")
+    - mkdir -p ~/.ssh && chmod 700 ~/.ssh
+    - 'echo "2048 SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8 github.com (RSA)" > github_sha_ref'
+    - 'ssh-keyscan -t rsa github.com | tee github-key-temp | ssh-keygen -lf - > github_sha_ssh'
+    - 'cmp --silent github_sha_ssh github_sha_ref || exit 1'
+    - 'cat github-key-temp >> ~/.ssh/known_hosts'
+    - git config --global user.email "noreply@icp.uni-stuttgart.de"
+    - git config --global user.name "espresso-ci"
+  script:
+    - cd "${HOME}"
+    - git clone -b main git@github.com:espressomd/espressomd.github.io.git
+    - cd espressomd.github.io
+    - LAST_COMMIT=$(git log -1 --pretty=format:"%s" -- doc dox tutorials)
+    - NEXT_COMMIT="Documentation for ${CI_COMMIT_SHA}"
+    - if [ "${NEXT_COMMIT}" = "${LAST_COMMIT}" ]; then echo "Documentation already up-to-date."; exit 0; fi
+    - rsync -a --delete --exclude='*.md5' --exclude='*.map' "${CI_PROJECT_DIR}/build/doc/doxygen/html/" dox/
+    - rsync -a --delete --exclude='.buildinfo' "${CI_PROJECT_DIR}/build/doc/sphinx/html/" doc/
+    - rsync -a --delete "${CI_PROJECT_DIR}/build/doc/tutorials/html/" tutorials/
+    - git add doc dox tutorials
+    - PREV_COMMIT_MSG=$(git log -1 --pretty=format:"%s")
+    - case "${PREV_COMMIT_MSG}" in "Documentation for "*) git reset --soft HEAD^; esac
+    - git commit --quiet -m "${NEXT_COMMIT}"
+    - git push -f origin main
+  interruptible: false
+  tags:
+    - docker
+    - linux
+    - icp
```

Comment:
I think we should start splitting our gitlab ci yaml files. This file get's too crowded.